test/test_utilities.cpp

```diff
@@ -85,6 +85,25 @@ void AssignInputValuesRandom(Tensor& A) {
   }
 }
 
+template <>
+// specialized template for float
+// randomly generate float data to populate the Tensor
+void AssignInputValuesRandom(Tensor& A, float min, float max) {
+  auto A_flat = A.flat<float>();
+  auto A_flat_data = A_flat.data();
+  srand(static_cast<unsigned>(time(0)));
+  for (int i = 0; i < A_flat.size(); i++) {
+    float value =
+        // randomly generate a number between 0 and (max-min)
+        static_cast<float>(rand()) / static_cast<float>(RAND_MAX / (max - min));
+    value = value + min;  // transform the range to min and max
+    value =
+        roundf(value * 100) / 100.0;  // change the precision of the float to
+                                      // 2 number after the decimal
+    A_flat_data[i] = value;
+  }
+}
+
```

Comment:
Not sure why we are specializing for float.
Compared to the unspecialized function this seems to be an extra line
```value = roundf(value * 100) / 100.0```
Why is this needed, and wouldn't the unspecialized one just work?

I might have missed some discussion why this is needed

Reply:
Precision adds some readability when debugging, the numbers are easier to comprehend. Easier to add 2.46 + 3.23 than 2.46343 + 3.23455 . This is one of the use cases  I can think of.

Reply:
but the main reason for adding this was the expectation that fusedbatchnormgrad test would pass, which it doesn't with this change.
Also we can always print stuff with reduced precision, rather than curtailing precision at the source.

Reply:
I meant when you are doing the math in your head to verify the expected results. Useful when the ops are simple. We can use the other utility functions for this purpose, i guess. 👍

Reply:
Hi @shresthamalik , @sayantan-nervana , some updates:

As I mentioned before, I added this restriction to solve the precision issue when testing FusedBatchNormGrad op.  

Now, @kanvi-nervana has a change for AssertTensorEquals() function where when comparing floats, she normalized the difference, and this exactly solves the precision problem we have seen before, so I also make a change for the FusedBatchNormGrad unit test to use the AssignInputValuesRandom() function and no decimal restriction is there.

In conclusion, the fundamental problem is solved by Kanvi's PR, and for my utility function, specialized template is not needed.